src/main/java/net/fabricmc/fabric/api/client/model/MultiRenderLayerBlock.java

```diff
@@ -16,6 +16,28 @@
 
 package net.fabricmc.fabric.api.client.model;
 
+import net.minecraft.block.Block;
+import net.minecraft.client.render.block.BlockRenderLayer;
+
+/**
+ * Interface implemented by {@link Block}s which wish to render on multiple chunk layers.
+ */
 public interface MultiRenderLayerBlock {
+	/**
+	 * Get a mask of additional render layers to use, supplementing the one provided
+	 * in {@link Block#getRenderLayer()}, in the form of "(1 << layer1.ordinal()) | (1 << layer2.ordinal()) | ...".
+	 *
+	 * Please note that getRenderLayer() should ideally be either:
+	 * (a) your primary rendering layer, with the additional layers used for "niceties",
+	 * (b) your "lower" (earlier in {@link BlockRenderLayer} order) rendering layers, as the one in said method
+	 * (the "primary" layer) will be used for rendering items.
+	 *
+	 * This is because:
+	 * (a) the hook for additional render layers is somewhat fragile and may not be immediately available
+	 * with a version update,
+	 * (b) the hook is only present for chunk rendering and is not, at present, supported for item rendering.
+	 *
+	 * @return The additional render layer mask.
+	 */
 	int getExtraRenderLayerMask();
 }
```

Comment:
This is clever but it also means the block has to know about its model.  It's better if the model tells you what layers it can render in.

The model should knows about the block and be designed to consume attributes the block will always provide, but the block shouldn't know how that information is being used.  The exception would be blocks that have a user-defined appearance or other behavior explicitly tied to rendering.

Example of how this would matter:  a resource pack maker wants to provide an alternate model for a block that is normally only rendered in the SOLID layer.  The alternate model also requires TRANSLUCENT. They want the solid layer to be emissive but the translucent layer isn't.   

In that scenario they could define the whole model as translucent, but that will be inefficient for translucency sort, and they won't get the lighting they want. Z-fighting might also be a problem.

Reply:
>This is clever but it also means the block has to know about its model. It's better if the model tells you what layers it can render in.

I've talked this over on IRC or elsewhere before, and the problem with that is, due to way Minecraft's rendering system is structured, it would be a noticeably more invasive (and slower!) patch to allow the *model* to decide.